lib/identity_cache/query_api.rb

```diff
@@ -228,7 +228,7 @@ def preload_id_embedded_associations(records)
           scope = child_model.all
           scope = scope.instance_exec(nil, &reflection.scope) if reflection.scope
 
-          pairs = scope.where(reflection.foreign_key => records.map(&:id)).pluck(reflection.foreign_key, reflection.active_record_primary_key)
+          pairs = scope.where(reflection.foreign_key => records.map(&:id)).pluck(reflection.foreign_key, child_model.primary_key)
```

Comment:
I am not sure I fully understand what you are saying, but for the newly create test case, we get following:
- `scope.klass` = `CustomChildRecord`
- `reflection.foreign_key` = `master_id`
- `reflection.active_record_primary_key` = `master_primary_key`

If we run the original code, it means something like `SELECT master_id, master_primary_key FROM custom_child_records`. This ends up with error, as `master_primary_key` column does not exist in the child table. Therefore I replaced it with `child_model.primary_key`, which is what I think should be here.

The reason it worked so far is in my opinion that it relied on the fact both the child and master tables used `id` as the primary key and therefore both `reflection.active_record_primary_key` and `child_model.primary_key` returned `id`.